src/parser/mod.rs

```diff
@@ -8353,20 +8409,40 @@ impl<'a> Parser<'a> {
         }
     }
 
+    /// Parse a possibly qualified, possibly quoted identifier, optionally allowing for wildcards,
+    /// e.g. *, `foo`.*, or "foo"."bar"
+    pub fn parse_object_name_with_wildcards(
```

Comment:
Still having a think about the best way to do this; posting the other updates for further feedback.

Reply:
Sorry it's taken me a while to get to this. It's a bit tricky, partly because the expression form shouldn't accept `*.*`. I will try to come back to this later, but have had higher priorities the last few weeks and probably will until later in January. In my opinion, it might be better to avoid dragging on with more merge conflicts etc. and merge without this suggested refactor. Let me know your thoughts.

Reply:
Ah not a problem I think this sounds reasonable! Could we maybe make the function non-public in case it needs to change later on so that we don't consider downstream impact?